cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -90,13 +100,16 @@ protected AbstractContext(Program program, RunId runId, Arguments arguments,
   protected AbstractContext(Program program, RunId runId, Arguments arguments,
                             Set<String> datasets, MetricsContext metricsContext,
                             DatasetFramework dsFramework, DiscoveryServiceClient discoveryServiceClient,
+                            LocationFactory locationFactory,
                             @Nullable AdapterDefinition adapterSpec,
-                            @Nullable PluginInstantiator pluginInstantiator) {
+                            @Nullable PluginInstantiator pluginInstantiator,
+                            @Nullable PluginInstantiator artifactPluginInstantiator) {
```

Comment:
o is pluginInstantiator for the template code path?  should add a todo to make it clear that its temporary